airflow/contrib/hooks/gcp_transfer_hook.py

```diff
@@ -56,52 +85,282 @@ def get_conn(self):
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def pause_transfer_operation(self, operation_name):
+        """
+        Pauses an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :rtype: None
+        """
+        self.get_conn()\
+            .transferOperations()\
+            .pause(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def resume_transfer_operation(self, operation_name):
+        """
+        Resumes an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :rtype: None
+        """
+        self.get_conn()\
+            .transferOperations()\
+            .resume(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def wait_for_transfer_job(self,
+                              job,
+                              expected_status=[GcpTransferOperationStatus.SUCCESS]):
+        """
+        Waits until the job reaches the expected state.
+
+        :param job: Transfer job
+        :type job: dict
+        :param expected_status: State that is expected
+            See:
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferOperations#Status
+        :type expected_status: string
+        :rtype: None
+        """
         while True:
-            result = self.get_conn().transferOperations().list(
-                name='transferOperations',
-                filter=json.dumps({
-                    'project_id': job['projectId'],
-                    'job_names': [job['name']],
-                }),
-            ).execute()
-            if self._check_operations_result(result):
-                return True
+            operations = self.list_transfer_operations(filter={
+                'project_id': job['projectId'],
+                'job_names': [job['name']]
+            })
+
+            if GCPTransferServiceHook.check_operations_statuses(operations, expected_status):
+                return
             time.sleep(TIME_TO_SLEEP_IN_SECONDS)
 
-    def _check_operations_result(self, result):
-        operations = result.get('operations', [])
+    def _append_project_id(self, body, alternative):
+        body = deepcopy(body)
+        body['project_id'] = body.get('project_id', self.project_id)
+        if not body['project_id']:
+            raise AirflowException("The project id must be passed either as "
+                                   "%s or as project_id extra in GCP "
+                                   "connection definition. Both are not set!"
+                                   % alternative)
+        return body
+
+    @staticmethod
+    def check_operations_statuses(operations, expected_statuses):
```

Comment:
We could maybe rename it to something like `is_operation_in_expected_status()` - this way invoking the function clearly states what it checks and what the return value will be (a boolean).
Currently it may be misleading when later on the method is invoked like:
```
if check_operations_statuses(...):
    return
```

Reply:
This method name was already earlier.  I'll think what we can do with it.

Reply:
I checked again. Earlier this method was private. Now, I change name of method to `is_operations_in_expected_statuses`.

Reply:
are_operations_in_expected_statuses